### ISSUE TITLE
[Request] Support the Electron Fuse: "EnableEmbeddedAsarIntegrityValidation"

### ISSUE THREAD

**Issue by Nantris** [electron-userland/electron-builder#6930]
See here for more info: https://github.com/electron/fuses/issues/7#issuecomment-1150505740

**Comment by kevinlinv** [comment]
+1, I think support for electron/fuses would be a great addition.

FYI, there was also a closed https://github.com/electron-userland/electron-builder/issues/6365 issue that request support.

**Comment by mmaietta** [comment]
Closing this one and reopening the other ticket.

**Comment by Slapbox** [comment]
From https://github.com/electron/fuses/issues/7#issuecomment-1150505740:

>>We're using Electron-Builder to package our app.

> That would be your problem sweat_smile. If you look at the official [@electron/asar](https://github.com/electron/asar/blob/main/lib/integrity.js) package, there is a module dedicated to generating these hashes. electron-builder, however, rolled their own ASAR creation code which doesn't have support for creating these hashes.

**Comment by rafaberaldo** [comment]
Would love to see this on electron-builder, Electron@30 added support for Windows machines as well.

**Comment by osztenkurden** [comment]
Is this currently on the roadmap? Given added support for Windows its kinda a big feature.

**Comment by mmaietta** [comment]
The only way to support this is by migrating to `@electron/asar` package. Unfortunately, I've tried to do so several times over the past few years and still haven't gotten to a solution that works. There's a lot of additional functionality that electron-builder relies on, such as transforming file data instead of copying a file directly. I can try and investigate this again, but can't promise anything. I'm the only maintainer on this project that remains (community contributions welcome!), so there's not really an "official" roadmap tbh, just trying to keep my head above the water with the combo of work+life outside of this project.

**Comment by mmaietta** [comment]
Quick update, this should officially be supported in an upcoming release once these two PRs are merged in for an alpha release
https://github.com/electron-userland/electron-builder/pull/8570 - electron/asar migration (what a PITA that was lol)
https://github.com/electron-userland/electron-builder/pull/8588 - electron/fuses integration

**Comment by mmaietta** [comment]
Released in [v26.0.0-alpha.2](https://github.com/electron-userland/electron-builder/releases/tag/v26.0.0-alpha.2). Please give it a shot!

Verified it locally (on mac build machine) that it launches with that fuse set:
```
npx @electron/fuses read --app dist/mac-universal/electron-quick-start-typescript.app
Analyzing app: electron-quick-start-typescript.app
Fuse Version: v1
  RunAsNode is Enabled
  EnableCookieEncryption is Disabled
  EnableNodeOptionsEnvironmentVariable is Enabled
  EnableNodeCliInspectArguments is Enabled
  EnableEmbeddedAsarIntegrityValidation is Enabled
  OnlyLoadAppFromAsar is Disabled
  LoadBrowserProcessSpecificV8Snapshot is Disabled
```

**Comment by osztenkurden** [comment]
Gonna test in a few hours, doing the gods work here, thank you! @mmaietta

**Comment by osztenkurden** [comment]
Hmm, while packaging works fine, I'm getting `Integrity check failed for asar archive` when trying to run the app

**Comment by mmaietta** [comment]
Hmmm yeah, from a quick read of asarmor, it's modifying the asar, which then causes the integrity generated/recorded by electron/asar to no longer be valid.

Can you try force setting `resetAdHocDarwinSignature` in `electronFuses` config to `true`?

Alternatively, you can play around with @electron/fuses in your afterPack via a convenience method I opened up as part of the integration for those that want to leverage `strictlyRequireAllFuses: true` to make sure their fuse config stays up to date.
```
const { FuseConfig, FuseVersion, FuseV1Options } = require("@electron/fuses")

exports.default = function (context: AfterPackContext) {
   // asarmor logic

  const fuses: FuseConfig = {
    version: FuseVersion.V1,
    resetAdHocDarwinSignature: true
    [FuseV1Options. EnableEmbeddedAsarIntegrityValidation]: true,
    ... // all other flags must be specified since `strictlyRequireAllFuses = true`
  }
  await context.packager.addElectronFuses(context, fuses)
}
```

I'm wondering if you need the `resetAdHocDarwinSignature` flag set.

**Comment by osztenkurden** [comment]
@mmaietta  That doesnt do the trick, maybe because Im on windows, but its not huge deal

**Comment by mmaietta** [comment]
Dam. Well it looks like asarmor is the issue as it manipulates the asar post electron/asar integrity generation since asarmor is injecting random bytes into the asar to prevent extraction.

An alternative approach could be using bytenode to obfuscate your code. It doesn't disable extraction like asarmor, but it doesn't require any asar manipulation either.

**Comment by osztenkurden** [comment]
Ill take a look, thanks!

**Comment by connor-orr1** [comment]
Hey @mmaietta, from reading the above, is `EnableEmbeddedAsarIntegrityValidation` only supported in the upcoming `v26.0.0` when packaging the apps via electron-builder?

Curiously enough I flipped the `EnableEmbeddedAsarIntegrityValidation` and `onlyLoadAppFromAsar` fuses when packaging with electron-builder `v25.0.0`, and when the checking the fuses for the app I tested, the app is loaded and the app doesn't crash.
Is this a potential false positive of the `EnableEmbeddedAsarIntegrityValidation` fuse being flipped given the changes to support the fuse are slated for release in `v26.0.0` as per the update you gave [here](https://github.com/electron-userland/electron-builder/issues/6930#issuecomment-2408253916)?
